### PR TITLE
Updating Chameleon Image handling documentation

### LINKED ISSUE #33647: ChameleonProcessor abstraction behaves unintuitively (ChameleonProcessor should also do VQ-GAE and BPE mapping and not ChameleonModel)
### Feature request

# Moving VQ-GAE and BPE mapping to ChameleonProcessor

I am relatively new to HuggingFace and the Transformers library, but have many years of ML and software engineering experience, so I expect there are probably legacy reasons for this unintuitive behavior, but I thought I would share my thoughts and offer to contribute the changes if you are interested.

## Background

Chameleon operates on a sequence of tokens, so as a user you need to supply the sequence of input tokens to make a prediction. But **getting the sequence of tokens is a bit involved**. For text we use a text tokenizer, and for images we use a sequence of transformations. Images undergo multiple transformations: center cropping, resizing, normalizing etc. Then this centered, cropped, normalized, image is fed to a Vector Quantized Variational AutoEncoder (VQ-GAE) to generate image tokens which are then mapped to Byte Pair Encoding (BPE) tokens.

Luckily, the ChameleonProcessor provides us with an easy way to make the tokens, or does it?

Intuitively the ChameleonProcessor should give me the sequence of tokens I need to feed to Chameleon....

## What does it give me?

The ChameleonProcessor when called on a string and list of images creates a dict with:
 

1. key "input_ids": mapped to the token sequence for the text and 1026 tokens for every instance of "\<image\>" in the original text.
2. key "pixel_values": mapped to the tensors of the pixel values for the centered, cropped, normalized images. 
3. key "attention_mask": not relevant to this discussion.

## What are these 1026 tokens for every "\<image\>"? 

They are duplicated placeholder image tokens (1024 replications of the "8711" token) plus two sentinel tokens (beginning and end of image)

I.e. the **VQ-GAE and BPE mapping for the images and the substitution is left to later with the call to the Chameleon model**.

## What would I expect:

I would expect the ChameleonProcessor to finish its task, i.e. to also apply the VQ-GAE, and the BPE mapping and give me back the **actual input_ids that will be passed to the model**.

I would expect that the input_ids sequence **contains the actual image tokens from the and not a sequence of 1024 sentinel tokens to be swapped in later based on the pixel_values** during the call to the ChameleonModel.

Alternatively there could be some kind of middle step that combines the "input_ids" and "pixel_values" and gives the actual "input_ids", those that will be passed to the Chameleon model.

## Benefits

Mostly this would be more intuitive to users, they would get the actual input_ids and not have any mysterious changing of the tokens in their model call

It would also bring some latency reductions to some applications as the model call would not need to reprocess all the images every time, I.e. you don't need to do the VQ-GAE, BPE, mapping, token replacement on every call which may actually be worthwhile.

### Motivation

I am doing a project on crafting adversarial attacks for Chameleon and found this behaviour unintuitive and thought it would be useful to fix it upstream for the world.

### Your contribution

I can do the coding if you are interested. If you are open to it I can have a chat with some of your staff and send it in

### PR DESCRIPTION
**Pull request by maxwbuckley**
# What does this PR do?

Improve Chameleon documentation. Especially focused on how the image processing works. 

Fixes #33647

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

### CODE REVIEW

File: docs/source/en/model_doc/chameleon.md [hunk 1]
```diff
@@ -54,8 +55,15 @@ The original code can be found [here](https://github.com/facebookresearch/chamel
...
+
+The images passed to `ChameleonProcessor` are processed through a sequence of transformations: format conversion, center cropping, resizing, normalizing etc. The processed images are put into the output dictionary as a tensor mapped to the key `pixel_values`.
+
+The conversion of the `pixel_values` tensor into image tokens and substitution into the placeholder tokens in `input_ids` happens when you call the `ChameleonModel`. This conversion is done in two steps, the first of which uses a [Vector Quantized Variational AutoEncoder (VQ-GAE)](https://arxiv.org/abs/1711.00937) to generate image tokens which are then mapped to Byte Pair Encoding (BPE) tokens. These image tokens are then swapped into the placeholder tokens in `input_ids`.
+
+Tip: If you want to see the image tokens a given image maps to, you can use the `ChameleonModel`'s `get_image_tokens` method, passing it the `pixel_values`.
```
Comment (zucchini-nlp) [inline docs/source/en/model_doc/chameleon.md:66]: Seems a bit too detailed, imo. Maybe we can wrap this in a "Note" section saying that image tokenization happens when calling the model and if one wants to obtain image tokens, then one should use `get_image_tokens()` method

Since the definition of what sis input ids and pixel values is already in model's docstring, this seems redundant.

**Review by zucchini-nlp — COMMENTED** [review]
Great! Thanks for making our docs better. It's good to make transparent what happens under the hood and how to get the `image tokens`, but for me seems too much redundant to explain what are ids and pixels. WDYT about the proposed change?

**Review by vignesh1507 — APPROVED** [review]
Thanks for updating the documentation.

**Comment by SUMiRE233** [comment]
Hi, I noticed this PR hasn’t been updated for a while. I’d be interested in helping refine the documentation based on the review feedback, especially making it more concise and possibly restructuring it into a clearer note section.

If that sounds good, I’d be happy to open a follow-up PR. I’m new to contributing to this repository, so any guidance would be appreciated. Thanks!

**Comment by hihry** [comment]
Hi, 
I have been intrigued with the discussion on Chameleon Image generation, I would be happy to create a PR on updating the documentation as suggested by @zucchini-nlp
